src/Libraries/Analysis/Utils.cs

```diff
@@ -17,5 +22,148 @@ internal static ColorRange1D CreateAnalyticalColorRange()
             var colorRange = ColorRange1D.ByColorsAndParameters(colors, parameters);
             return colorRange;
         }
+
+        public static double[][] CreateGradientValueMap(IEnumerable<double> values, IEnumerable<UV> locations, int uSamples, int vSamples)
+        {
+            if (!values.Any())
+            {
+                return new double[0][];
+            }
+
+            // Zip the values and locations together to create an
+            // enumerable of ValueLocation objects.
+            var zip = locations.Zip(values, (uv, value) => new ValueLocation<double>(value, uv));
+
+            // The size of the map will be +1 in both directions
+            // than the number of samples, to ensure that we are
+            // sampling all the way to the edges.
+
+            var w = uSamples + 1;
+            var h = vSamples + 1;
+
+            var valueMap = new double[w][];
+            for (var i = 0; i < w; i++)
+            {
+                valueMap[i] = new double[h];
+            }
+
+            Parallel.For(0,w,
```

Comment:
Gradient calculations are done in parallel. We should eventually move gradient calculation to the GPU.